net/mwan3/files/usr/sbin/mwan3rtmon

```diff
@@ -107,14 +107,20 @@ mwan3_rtmon_route_handle()
 		local iface=$1
 		tbl=$($IP route list table $tid 2>/dev/null)$'\n'
 
-		if [ "$(cat /var/run/mwan3track/$iface/STATUS)" != "online" ]; then
+		if [ -n "$iface" ] && [ "$(cat /var/run/mwan3track/$iface/STATUS)" != "online" ]; then
 			LOG debug "interface $iface is offline - skipping $route_line";
 			return
 		fi
 
 		# check that action needs to be performed. May not need to take action if we
 		# got a delete event, but table was already flushed
-		if [ $action = "del" ] && [ -n "${tbl##*$route_line$'\n'*}" ]; then
+
+		# need to test with and without trailing space because some `iproute2` versions
+		# output a trailing space
+
+		if [ $action = "del" ] &&
+			   [ -n "${tbl##*$route_line$'\n'*}" ] &&
+			   [ -n "${tbl##*$route_line$' \n'*}" ]; then
```

Comment:
You don't seems to get my point in my P/R for this bug, where in https://github.com/openwrt/packages/pull/13756/commits/ef853528dec0a4b89a4ead0039c252558d3ff663

what the problem is
`"${tbl##*$route_line$'\n'*}"` is diff from `${tbl##*$route_line$'\n'*}`
that is why you get wrong test resault on this line

and more changes on https://github.com/openwrt/packages/pull/13756/commits/ef853528dec0a4b89a4ead0039c252558d3ff663

you could see the detail, and I am sure my changes resolve the real problem.

Reply:
reload trigger on network cannot be accepted, I think.